src/track/trackref.cpp

```diff
@@ -0,0 +1,43 @@
+#include "track/trackref.h"
+
+
+bool TrackRef::verifyConsistency() const {
+    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasCanonicalLocation()) {
+        return false;
+    }
+    DEBUG_ASSERT_AND_HANDLE(hasLocation() || !hasId()) {
+        return false;
+    }
+    return true;
+}
+
+QFileInfo TrackRef::createFileInfo() const {
+    QFileInfo fileInfo(getLocation());
```

Comment:
Is this a performance penalty? If yes, does it make sense to cache the object? 
It is probably hard because of concurrency ....
Thoughts?